### ISSUE TITLE
Fix errors reported by TypeScript 5.2

### ISSUE THREAD

**Issue by regseb** [markuplint/markuplint#1124]
- Markuplint version: `3.12.0`
- Node.js version: `v20.5.1`
- OS: Ubuntu 22.04.3 LTS

## Describe the bug

TypeScript reports errors in Markuplint. The problem has been present since TypeScript version _5.2_. I had a similar issue **https://github.com/markuplint/markuplint/issues/1061** when I upgraded from version _5.0_ to _5.1_: TypeScript had added new methods to the DOM interfaces. Perhaps this problem should be solved by replacing `MLElement` with `MLNode` in [`after()`, `before()` and `replaceWith()`](https://github.com/microsoft/TypeScript/blob/e69a0b0e38b380677adb83ff0d20d8b8c32b2d9d/src/lib/dom.generated.d.ts#L5571-L5602):

```diff
-after(...nodes: (string | MLElement<any, any>)[]): void;
+after(...nodes: (string | MLNode<any, any>)[]): void;

-before(...nodes: (string | MLElement<any, any>)[]): void;
+before(...nodes: (string | MLNode<any, any>)[]): void;

-replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
+replaceWith(...nodes: (string | MLNode<any, any>)[]): void;
```

## Code Example

- `package.json`

  ```JSON
  {
      "name": "testcase",
      "version": "1.0.0",
      "private": true,
      "dependencies": {
          "markuplint": "3.12.0",
          "typescript": "5.2.2"
      }
  }
  ```

- `index.ts`

  ```TypeScript
  import { MLEngine } from "markuplint";

  console.log(MLEngine);
  ```
## Steps To Reproduce

1. `npm install`
2. `npx tsc --lib esnext,dom --target esnext --moduleResolution nodenext --module nodenext index.ts`

## The current behavior

```
node_modules/@markuplint/ml-core/lib/ml-dom/node/attr.d.ts:21:14 - error TS2416: Property 'ownerElement' in type 'MLAttr<T, O>' is not assignable to the same property in base type 'Attr'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.
    The types of 'attributes.getNamedItem(...).childNodes' are incompatible between these types.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
        Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
          Type 'MLElement<T, O>' is not assignable to type 'ChildNode'.
            Types of property 'after' are incompatible.
              Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

21     readonly ownerElement: MLElement<T, O>;
                ~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:26:9 - error TS2416: Property 'nextElementSibling' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.
    The types of 'attributes.getNamedItem(...).childNodes' are incompatible between these types.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
        Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
          Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
            Types of property 'after' are incompatible.
              Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
                Types of parameters 'nodes' and 'nodes' are incompatible.
                  Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.
                    Type 'Node' is not assignable to type 'string | MLElement<any, any>'.
                      Type 'Node' is missing the following properties from type 'MLElement<any, any>': #private, closeTag, elementType, endSpace, and 288 more.

26     get nextElementSibling(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:35:9 - error TS2416: Property 'previousElementSibling' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

35     get previousElementSibling(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:39:5 - error TS2416: Property 'after' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

39     after(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:44:5 - error TS2416: Property 'before' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

44     before(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:55:5 - error TS2416: Property 'replaceWith' in type 'MLCharacterData<T, O, A>' is not assignable to the same property in base type 'CharacterData'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

55     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/comment.d.ts:5:22 - error TS2420: Class 'MLComment<T, O>' incorrectly implements interface 'Comment'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

5 export declare class MLComment<T extends RuleConfigValue, O extends PlainData = undefined> extends MLCharacterData<T, O, MLASTComment> implements Comment {
                       ~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-fragment.d.ts:5:22 - error TS2420: Class 'MLDocumentFragment<T, O>' incorrectly implements interface 'DocumentFragment'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

5 export declare class MLDocumentFragment<T extends RuleConfigValue, O extends PlainData = undefined> extends MLParentNode<T, O, MLASTAbstractNode> implements DocumentFragment {
                       ~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:25:5 - error TS2416: Property 'after' in type 'MLDocumentType<T, O>' is not assignable to the same property in base type 'DocumentType'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

25     after(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:29:5 - error TS2416: Property 'before' in type 'MLDocumentType<T, O>' is not assignable to the same property in base type 'DocumentType'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

29     before(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:37:5 - error TS2416: Property 'replaceWith' in type 'MLDocumentType<T, O>' is not assignable to the same property in base type 'DocumentType'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

37     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
       ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:206:9 - error TS2416: Property 'doctype' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type 'MLDocumentType<T, O>' is not assignable to type 'DocumentType'.
    Types of property 'childNodes' are incompatible.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
        Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
          Type 'MLElement<T, O>' is not assignable to type 'ChildNode'.
            Types of property 'after' are incompatible.
              Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

206     get doctype(): MLDocumentType<T, O> | null;
            ~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1441:5 - error TS2416: Property 'getElementById' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type '(elementId: string) => MLElement<T, O>' is not assignable to type '(elementId: string) => HTMLElement'.
    Type 'MLElement<T, O>' is missing the following properties from type 'HTMLElement': popover, hidePopover, showPopover, togglePopover, onscrollend

1441     getElementById(elementId: string): MLElement<T, O> | null;
         ~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1448:5 - error TS2416: Property 'getElementsByClassName' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type '(classNames: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '(classNames: string) => HTMLCollectionOf<Element>'.
    Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollectionOf<Element>'.
      Type 'MLElement<T, O>' is not assignable to type 'Element'.
        The types of 'attributes.getNamedItem(...).childNodes' are incompatible between these types.
          Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.

1448     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1448:66 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.

1448     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
                                                                      ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1462:5 - error TS2416: Property 'getElementsByTagName' in type 'MLDocument<T, O>' is not assignable to the same property in base type 'Document'.
  Type '(qualifiedName: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '{ <K extends keyof HTMLElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<HTMLElementTagNameMap[K]>; <K extends keyof SVGElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<...>; <K extends keyof MathMLElementTagNameMap>(qualifiedName: K): HTMLCollectionOf<...>; <K extends keyof HTMLElementDeprecatedTagNameMa...'.
    Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollectionOf<Element>'.

1462     getElementsByTagName(qualifiedName: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:1462:67 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.

1462     getElementsByTagName(qualifiedName: string): HTMLCollectionOf<MLElement<T, O>>;
                                                                       ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:369:9 - error TS2416: Property 'attributes' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type 'MLNamedNodeMap<T, O>' is not assignable to type 'NamedNodeMap'.
    The types of 'getNamedItem(...).ownerElement.insertAdjacentElement' are incompatible between these types.
      Type '(where: InsertPosition, element: MLElement<T, O>) => MLElement<T, O>' is not assignable to type '(where: InsertPosition, element: Element) => Element'.
        Types of parameters 'element' and 'element' are incompatible.
          Type 'Element' is missing the following properties from type 'MLElement<T, O>': #private, closeTag, elementType, endSpace, and 170 more.

369     get attributes(): MLNamedNodeMap<T, O>;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:369:9 - error TS2416: Property 'attributes' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type 'MLNamedNodeMap<T, O>' is not assignable to type 'NamedNodeMap'.

369     get attributes(): MLNamedNodeMap<T, O>;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:557:9 - error TS2416: Property 'nextElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

557     get nextElementSibling(): MLElement<T, O> | null;
            ~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:557:9 - error TS2416: Property 'nextElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

557     get nextElementSibling(): MLElement<T, O> | null;
            ~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1391:9 - error TS2416: Property 'previousElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

1391     get previousElementSibling(): MLElement<T, O> | null;
             ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1391:9 - error TS2416: Property 'previousElementSibling' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

1391     get previousElementSibling(): MLElement<T, O> | null;
             ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1515:5 - error TS2416: Property 'after' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

1515     after(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1515:5 - error TS2416: Property 'after' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

1515     after(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1543:5 - error TS2416: Property 'before' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

1543     before(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1543:5 - error TS2416: Property 'before' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

1543     before(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1623:5 - error TS2416: Property 'getAttributeNode' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.
    Type 'MLAttr<T, O>' is not assignable to type 'Attr'.

1623     getAttributeNode(qualifiedName: string): MLAttr<T, O> | null;
         ~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1623:5 - error TS2416: Property 'getAttributeNode' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.

1623     getAttributeNode(qualifiedName: string): MLAttr<T, O> | null;
         ~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1665:5 - error TS2416: Property 'getElementsByClassName' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(classNames: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '(classNames: string) => HTMLCollectionOf<Element>'.
    Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollectionOf<Element>'.
      Type 'MLElement<T, O>' is not assignable to type 'Element'.

1665     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1665:5 - error TS2416: Property 'getElementsByClassName' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(classNames: string) => HTMLCollectionOf<MLElement<T, O>>' is not assignable to type '(classNames: string) => HTMLCollectionOf<Element>'.

1665     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
         ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1665:66 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.

1665     getElementsByClassName(classNames: string): HTMLCollectionOf<MLElement<T, O>>;
                                                                      ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1730:5 - error TS2416: Property 'insertAdjacentElement' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(where: InsertPosition, element: MLElement<T, O>) => MLElement<T, O>' is not assignable to type '(where: InsertPosition, element: Element) => Element'.
    Types of parameters 'element' and 'element' are incompatible.
      Type 'Element' is not assignable to type 'MLElement<T, O>'.

1730     insertAdjacentElement(where: InsertPosition, element: MLElement<T, O>): MLElement<T, O> | null;
         ~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1730:5 - error TS2416: Property 'insertAdjacentElement' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(where: InsertPosition, element: MLElement<T, O>) => MLElement<T, O>' is not assignable to type '(where: InsertPosition, element: Element) => Element'.

1730     insertAdjacentElement(where: InsertPosition, element: MLElement<T, O>): MLElement<T, O> | null;
         ~~~~~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1803:5 - error TS2416: Property 'replaceWith' in type 'MLElement<T, O>' is not assignable to the same property in base type 'Element'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
    Types of parameters 'nodes' and 'nodes' are incompatible.
      Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

1803     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:1803:5 - error TS2416: Property 'replaceWith' in type 'MLElement<T, O>' is not assignable to the same property in base type 'HTMLElement'.
  Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

1803     replaceWith(...nodes: (string | MLElement<any, any>)[]): void;
         ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:4:5 - error TS2416: Property 'getNamedItem' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.
    Call signature return types 'MLAttr<T, O>' and 'Attr' are incompatible.
      The types of 'ownerElement.attributes.setNamedItem' are incompatible between these types.
        Type '(attr: MLAttr<T, O>) => MLAttr<T, O>' is not assignable to type '(attr: Attr) => Attr'.
          Types of parameters 'attr' and 'attr' are incompatible.
            Type 'Attr' is missing the following properties from type 'MLAttr<T, O>': #private, endQuote, equal, isDuplicatable, and 33 more.

4     getNamedItem(qualifiedName: string): MLAttr<T, O> | null;
      ~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:12:5 - error TS2416: Property 'item' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(index: number) => MLAttr<T, O>' is not assignable to type '(index: number) => Attr'.
    Type 'MLAttr<T, O>' is not assignable to type 'Attr'.

12     item(index: number): MLAttr<T, O> | null;
       ~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:19:5 - error TS2416: Property 'removeNamedItem' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(qualifiedName: string) => MLAttr<T, O>' is not assignable to type '(qualifiedName: string) => Attr'.
    Type 'MLAttr<T, O>' is not assignable to type 'Attr'.

19     removeNamedItem(qualifiedName: string): MLAttr<T, O>;
       ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:26:5 - error TS2416: Property 'removeNamedItemNS' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(namespace: string, localName: string) => MLAttr<T, O>' is not assignable to type '(namespace: string, localName: string) => Attr'.
    Type 'MLAttr<T, O>' is not assignable to type 'Attr'.

26     removeNamedItemNS(namespace: string | null, localName: string): MLAttr<T, O>;
       ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:33:5 - error TS2416: Property 'setNamedItem' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(attr: MLAttr<T, O>) => MLAttr<T, O>' is not assignable to type '(attr: Attr) => Attr'.

33     setNamedItem(attr: MLAttr<T, O>): MLAttr<T, O> | null;
       ~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:40:5 - error TS2416: Property 'setNamedItemNS' in type 'MLNamedNodeMap<T, O>' is not assignable to the same property in base type 'NamedNodeMap'.
  Type '(attr: MLAttr<T, O>) => MLAttr<T, O>' is not assignable to type '(attr: Attr) => Attr'.
    Types of parameters 'attr' and 'attr' are incompatible.
      Type 'Attr' is not assignable to type 'MLAttr<T, O>'.

40     setNamedItemNS(attr: MLAttr<T, O>): MLAttr<T, O> | null;
       ~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:131:9 - error TS2416: Property 'childNodes' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
    Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
      Type 'MLDocumentType<T, O>' is not assignable to type 'ChildNode'.
        Types of property 'after' are incompatible.
          Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

131     get childNodes(): NodeListOf<MLChildNode<T, O>>;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:131:34 - error TS2344: Type 'MLChildNode<T, O>' does not satisfy the constraint 'Node'.
  Type 'MLDocumentType<T, O>' is not assignable to type 'Node'.
    Types of property 'childNodes' are incompatible.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.

131     get childNodes(): NodeListOf<MLChildNode<T, O>>;
                                     ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:139:9 - error TS2416: Property 'firstChild' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.

139     get firstChild(): MLChildNode<T, O> | null;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:155:9 - error TS2416: Property 'lastChild' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.

155     get lastChild(): MLChildNode<T, O> | null;
            ~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:208:9 - error TS2416: Property 'parentElement' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLElement<T, O>' is missing the following properties from type 'HTMLElement': popover, hidePopover, showPopover, togglePopover, onscrollend

208     get parentElement(): MLElement<T, O> | null;
            ~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:256:9 - error TS2416: Property 'parentNode' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type 'MLElement<T, O> | MLDocument<any, any> | MLDocumentFragment<any, any>' is not assignable to type 'ParentNode'.
    Type 'MLElement<T, O>' is not assignable to type 'ParentNode'.
      The types of 'children.namedItem(...).attributes.getNamedItem(...).childNodes' are incompatible between these types.
        Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.

256     get parentNode(): MLDocument<any, any> | MLDocumentFragment<any, any> | MLElement<T, O> | null;
            ~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:381:5 - error TS2416: Property 'getRootNode' in type 'MLNode<T, O, A>' is not assignable to the same property in base type 'Node'.
  Type '(options?: GetRootNodeOptions) => MLNode<T, O, MLASTAbstractNode>' is not assignable to type '(options?: GetRootNodeOptions) => Node'.
    Call signature return types 'MLNode<T, O, MLASTAbstractNode>' and 'Node' are incompatible.
      The types of 'childNodes' are incompatible between these types.
        Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.

381     getRootNode(options?: GetRootNodeOptions): MLNode<T, O>;
        ~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:481:43 - error TS2344: Type 'MLChildNode<T, O>' does not satisfy the constraint 'Node'.

481     resetChildren(childNodes?: NodeListOf<MLChildNode<T, O>>): void;
                                              ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:20:9 - error TS2416: Property 'children' in type 'MLParentNode<T, O, A>' is not assignable to the same property in base type 'ParentNode'.
  Type 'HTMLCollectionOf<MLElement<T, O>>' is not assignable to type 'HTMLCollection'.
    The types of 'namedItem(...).attributes.getNamedItem(...).childNodes' are incompatible between these types.
      Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
        Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
          Type 'MLElement<T, O>' is not assignable to type 'ChildNode'.
            Types of property 'after' are incompatible.
              Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.
                Types of parameters 'nodes' and 'nodes' are incompatible.
                  Type 'string | Node' is not assignable to type 'string | MLElement<any, any>'.

20     get children(): HTMLCollectionOf<MLElement<T, O>>;
           ~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:20:38 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Element'.

20     get children(): HTMLCollectionOf<MLElement<T, O>>;
                                        ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:25:9 - error TS2416: Property 'firstElementChild' in type 'MLParentNode<T, O, A>' is not assignable to the same property in base type 'ParentNode'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

25     get firstElementChild(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:30:9 - error TS2416: Property 'lastElementChild' in type 'MLParentNode<T, O, A>' is not assignable to the same property in base type 'ParentNode'.
  Type 'MLElement<T, O>' is not assignable to type 'Element'.

30     get lastElementChild(): MLElement<T, O> | null;
           ~~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:56:53 - error TS2344: Type 'MLElement<T, O>' does not satisfy the constraint 'Node'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.

56     querySelectorAll(selectors: string): NodeListOf<MLElement<T, O>>;
                                                       ~~~~~~~~~~~~~~~

node_modules/@markuplint/ml-core/lib/ml-dom/node/text.d.ts:5:22 - error TS2420: Class 'MLText<T, O>' incorrectly implements interface 'Text'.
  Types of property 'childNodes' are incompatible.
    Type 'NodeListOf<MLChildNode<T, O>>' is not assignable to type 'NodeListOf<ChildNode>'.
      Type 'MLChildNode<T, O>' is not assignable to type 'ChildNode'.
        Type 'MLText<T, O>' is not assignable to type 'ChildNode'.
          Types of property 'after' are incompatible.
            Type '(...nodes: (string | MLElement<any, any>)[]) => void' is not assignable to type '(...nodes: (string | Node)[]) => void'.

5 export declare class MLText<T extends RuleConfigValue, O extends PlainData = undefined> extends MLCharacterData<T, O, MLASTText> implements Text {
                       ~~~~~~


Found 56 errors in 11 files.

Errors  Files
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/attr.d.ts:21
     5  node_modules/@markuplint/ml-core/lib/ml-dom/node/character-data.d.ts:26
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/comment.d.ts:5
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/document-fragment.d.ts:5
     3  node_modules/@markuplint/ml-core/lib/ml-dom/node/document-type.d.ts:25
     6  node_modules/@markuplint/ml-core/lib/ml-dom/node/document.d.ts:206
    19  node_modules/@markuplint/ml-core/lib/ml-dom/node/element.d.ts:369
     6  node_modules/@markuplint/ml-core/lib/ml-dom/node/named-node-map.d.ts:4
     8  node_modules/@markuplint/ml-core/lib/ml-dom/node/node.d.ts:131
     5  node_modules/@markuplint/ml-core/lib/ml-dom/node/parent-node.d.ts:20
     1  node_modules/@markuplint/ml-core/lib/ml-dom/node/text.d.ts:5
```

## The expected behavior

No errors.

**Comment by YusukeHirao** [comment]
@regseb Thank you! I'll fix and update it.
